src/main/java/com/example/backend/Model/City.java

```diff
@@ -0,0 +1,84 @@
+package com.example.backend.Model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class City {
+	private double lat;
+	private double lon;
+	private String name;
+	
+	private static ArrayList<City> list = new ArrayList<>(
+			Arrays.asList(new City("Barcelona", 41.346176, 2.168365),
+				new City("Tokyo", 35.710064, 139.810699),
+				new City("Osaka", 34.672314, 135.484802),
+				new City("San Jose", 37.335480, -121.893028),
+				new City("New York City", 40.730610, -73.935242)));
+	
+	public City() { 
+		//empty constructor
+	}
+	
+	public City(String name, double lat, double lon) {
+		this.name = name;
+		this.lat = lat;
+		this.lon = lon;
+	}
+	
+	public String getCityName() {
+		return name;
+	}
+	
+	public double getLatitude() {
+		return lat;
+	}
+	
+	public double getLongitude() {
+		return lon;
+	}
+	
+	public void setCityName(String name) {
+		this.name = name;
+	}
+	
+	public void setLatitude(double lat) {
+		this.lat = lat;
+	}
+	
+	public void setLongitude(double lon) {
+		this.lon = lon;
+	}
+	
+	public static void addCity(String name, double lat, double lon) {
+		list.add(new City(name, lat, lon));
+	}
+	
+	public static City findNearest(double lat, double lon) {
+		double min = Double.MAX_VALUE;
+		City minCity = null;
+		
+		for (City c: list) {
+			double dist = haversineDistance(lat, lon, c.getLatitude(), c.getLongitude());
+			if (dist < min) {
+				min = dist;
+				minCity = c;
+			}
+		}
+		return minCity;
+	}
+	
+	private static double haversineDistance(double fromLat, double fromLon, double lat, double lon) {
```

Comment:
what is haversineDistance? Can you include the formula link that you used for this?

Reply:
I got the formula from Wikipedia: https://en.wikipedia.org/wiki/Haversine_formula. It's used to find the distance between two points using latitude and longitude.